src/Model/Item.php

```diff
@@ -1528,6 +1528,29 @@ public static function guidFromUri($uri, $host)
 		return $guid_prefix.$host_hash;
 	}
 
+	/**
+	 * generate an unique URI
+	 *
+	 * @param integer $uid User id
+	 * @param string $guid An existing GUID (Otherwise it will be generated)
+	 *
+	 * @return string
+	 */
+	public static function newURI($uid, $guid = "")
+	{
+		if ($guid == "") {
+			$guid = get_guid(32);
```

Comment:
Why didn't you keep the existence check?

Reply:
It had been from a time where there could have been identical GUIDs. We are now using a standard method for creating unique values.